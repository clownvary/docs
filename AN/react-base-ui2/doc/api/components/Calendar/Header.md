## Header 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
onPrevClick|func|&ensp;|true|&ensp;
onNextClick|func|&ensp;|true|&ensp;
onTitleClick|func|&ensp;|true|&ensp;
onTodayClick|func|&ensp;|true|&ensp;
title|string|&ensp;|true|&ensp;
displayToday|bool|&ensp;|true|&ensp;
todayLabel|string|&ensp;|true|&ensp;
prefix|string|&ensp;|true|&ensp;