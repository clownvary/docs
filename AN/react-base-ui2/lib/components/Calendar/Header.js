'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');var _propTypes2 = _interopRequireDefault(_propTypes);

var _consts = require('../../consts');
var _utils = require('../../utils');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var HeaderPropTypes = {
  onPrevClick: _propTypes2.default.func.isRequired,
  onNextClick: _propTypes2.default.func.isRequired,
  onTitleClick: _propTypes2.default.func.isRequired,
  onTodayClick: _propTypes2.default.func.isRequired,

  title: _propTypes2.default.string.isRequired,
  displayToday: _propTypes2.default.bool.isRequired,
  todayLabel: _propTypes2.default.string.isRequired,
  prefix: _propTypes2.default.string.isRequired };var


Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'render', value: function render()



    {var _this2 = this;
      return (
        _react2.default.createElement('div', { className: this.props.prefix + 'calendar-header' },
          _react2.default.createElement('i', {
            className: 'icon icon-chevron-left',
            onClick: this.props.onPrevClick,
            tabIndex: 0,
            onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
              [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
              function () {return _this2.props.onPrevClick();});} }),



          this.props.displayToday &&
          _react2.default.createElement('span', {
              className: this.props.prefix + 'calendar-header-today',
              onClick: this.props.onTodayClick,
              tabIndex: 0,
              onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
                [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
                function () {return _this2.props.onTodayClick();});} },

            this.props.todayLabel),

          _react2.default.createElement('span', {
              className: this.props.prefix + 'calendar-header-title',
              onClick: this.props.onTitleClick,
              tabIndex: 0,
              onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
                [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
                function () {return _this2.props.onTitleClick();});} },

            this.props.title),
          _react2.default.createElement('i', {
            className: 'icon icon-chevron-right',
            onClick: this.props.onNextClick,
            tabIndex: 0,
            onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
              [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
              function () {return _this2.props.onNextClick();});} })));




    } }]);return Header;}(_react2.default.PureComponent);Header.displayName = 'Header';Header.propTypes = HeaderPropTypes;exports.default =


Header;module.exports = exports['default'];