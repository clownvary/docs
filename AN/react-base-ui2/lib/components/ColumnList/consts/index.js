'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ColumnType = require('./ColumnType');var _ColumnType2 = _interopRequireDefault(_ColumnType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);
var _SortOrder = require('../../../consts/SortOrder');var _SortOrder2 = _interopRequireDefault(_SortOrder);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  SortOrder: _SortOrder2.default,
  ColumnType: _ColumnType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];