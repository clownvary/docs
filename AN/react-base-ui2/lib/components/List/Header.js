'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');var _propTypes2 = _interopRequireDefault(_propTypes);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);

var _Input = require('../Input');var _Input2 = _interopRequireDefault(_Input);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var HeaderPropTypes = {
  config: _propTypes2.default.shape({
    disabled: _propTypes2.default.bool,
    filterable: _propTypes2.default.bool // can be filter by inputted value
  }),

  onFilter: _propTypes2.default.func };var


Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'renderFilter', value: function renderFilter()



    {var _props =
      this.props,_props$config = _props.config,disabled = _props$config.disabled,filterable = _props$config.filterable,onFilter = _props.onFilter;
      return filterable && (0, _isFunction2.default)(onFilter) &&
      _react2.default.createElement(_Input2.default, {
        preIcon: 'icon icon-search',
        disabled: disabled,
        onChange: function onChange(e) {return onFilter({ filter: e.target.value });} });

    } }, { key: 'render', value: function render()

    {var _props2 =
      this.props,_props2$config = _props2.config,prefix = _props2$config.prefix,filterable = _props2$config.filterable,onFilter = _props2.onFilter;

      var isHeaderEmpty = !(filterable && (0, _isFunction2.default)(onFilter));
      return (
        !isHeaderEmpty ?
        _react2.default.createElement('div', { className: (0, _classnames2.default)(prefix + 'list__header', { hidden: isHeaderEmpty }) },
          this.renderFilter()) :
        null);

    } }]);return Header;}(_react2.default.PureComponent);Header.displayName = 'Header';Header.propsType = HeaderPropTypes;exports.default =


Header;module.exports = exports['default'];