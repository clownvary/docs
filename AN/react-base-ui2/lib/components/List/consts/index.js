'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ListType = require('./ListType');var _ListType2 = _interopRequireDefault(_ListType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  ListType: _ListType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];