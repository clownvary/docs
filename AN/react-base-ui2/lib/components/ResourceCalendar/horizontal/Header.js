'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var

Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'onDateHeaderClick', value: function onDateHeaderClick(
    e, date) {
      e.stopPropagation();var

      onDateHeaderClick = this.props.onDateHeaderClick;
      if ((0, _isFunction2.default)(onDateHeaderClick)) {
        onDateHeaderClick(e, date);
      }
    } }, { key: 'renderCell', value: function renderCell(

    date) {var _this2 = this;var _props =
      this.props,_props$dateFormat = _props.dateFormat,dateFormat = _props$dateFormat === undefined ? 'DD ddd' : _props$dateFormat,currentDate = _props.currentDate,onDateHeaderClick = _props.onDateHeaderClick;
      var isToday = currentDate ? currentDate.isSame(date.value, 'day') : false;
      var isWeekend = date.value.day() === 0 || date.value.day() === 6;
      var isSunday = date.value.day() === 0;
      var classes = (0, _classnames2.default)(
      'grid-cell an-rc-date-header',
      {
        today: isToday,
        weekend: isWeekend,
        'is-sunday': isSunday });

      /* istanbul ignore next */
      var text = date ? date.value.format(dateFormat) : '';
      return (
        _react2.default.createElement('th', {
            className: classes,
            key: 'header_' + date.key },

          _react2.default.createElement('span', {
              className: (0, _classnames2.default)({ 'an-rc-clickable': (0, _isFunction2.default)(onDateHeaderClick) }),
              onClick: function onClick(e) {return _this2.onDateHeaderClick(e, date.value);} },

            isToday ? 'Today' : text)));



    } }, { key: 'render', value: function render()

    {var _this3 = this;var _props$dates =
      this.props.dates,dates = _props$dates === undefined ? [] : _props$dates;
      return (
        _react2.default.createElement('table', { className: 'an-rc-grid an-rc-grid-header' },
          _react2.default.createElement('thead', null,
            _react2.default.createElement('tr', { className: 'grid-row' },

              dates.map(function (date) {return _this3.renderCell(date);})))));





    } }]);return Header;}(_react2.default.PureComponent);exports.default =


Header;module.exports = exports['default'];