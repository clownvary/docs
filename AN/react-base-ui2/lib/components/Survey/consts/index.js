'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _QuestionType = require('./QuestionType');var QuestionType = _interopRequireWildcard(_QuestionType);
var _InputType = require('./InputType');var InputType = _interopRequireWildcard(_InputType);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}exports.default =

{
  QuestionType: QuestionType,
  InputType: InputType };module.exports = exports['default'];