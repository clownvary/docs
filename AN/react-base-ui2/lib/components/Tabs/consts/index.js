'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _TabsClasses = require('./TabsClasses');var _TabsClasses2 = _interopRequireDefault(_TabsClasses);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =


{
  TabsClasses: _TabsClasses2.default };module.exports = exports['default'];