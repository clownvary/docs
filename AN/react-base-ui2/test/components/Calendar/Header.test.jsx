import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from 'src/components/Calendar/Header';
import { KeyCode } from 'src/consts';

describe('components/Calendar/Header', () => {
  it('CalendarHeader renders fine', () => {
    const snapshot = renderer.create(
      <Header
        title={'2018-06-05'}
        prefix={'snapshot-calendar-prefix-'}
        displayToday
        todayLabel={'Today'}
        onPrevClick={jest.fn()}
        onTodayClick={jest.fn()}
        onTitleClick={jest.fn()}
        onNextClick={jest.fn()}
      />
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('CalendarHeader works fine', () => {
    const props = {
      prefix: 'ut-',
      displayToday: true,
      title: 'Header',
      todayLabel: 'Today',
      onPrevClick: jest.fn(),
      onTodayClick: jest.fn(),
      onTitleClick: jest.fn(),
      onNextClick: jest.fn()
    };
    const header = shallow(<Header {...props} />);

    const prev = header.find('i.icon-chevron-left');
    expect(prev).toHaveLength(1);
    prev.simulate('keydown', { keyCode: KeyCode.ENTER, preventDefault: () => {} });
    expect(props.onPrevClick).toHaveBeenCalledTimes(1);

    const today = header.find(`span.${props.prefix}calendar-header-today`);
    expect(today).toHaveLength(1);
    today.simulate('keydown', { keyCode: KeyCode.SPACE, preventDefault: () => {} });
    expect(props.onTodayClick).toHaveBeenCalledTimes(1);

    const title = header.find(`span.${props.prefix}calendar-header-title`);
    expect(title).toHaveLength(1);
    title.simulate('keydown', { keyCode: KeyCode.SPACE, preventDefault: () => {} });
    expect(props.onTitleClick).toHaveBeenCalledTimes(1);

    const next = header.find('i.icon-chevron-right');
    expect(next).toHaveLength(1);
    next.simulate('keydown', { keyCode: KeyCode.ENTER, preventDefault: () => {} });
    expect(props.onNextClick).toHaveBeenCalledTimes(1);
  });
});
