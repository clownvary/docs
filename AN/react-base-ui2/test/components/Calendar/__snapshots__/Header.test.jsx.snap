// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Calendar/Header CalendarHeader renders fine 1`] = `
<div
  className="snapshot-calendar-prefix-calendar-header"
>
  <i
    className="icon icon-chevron-left"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  />
  <span
    className="snapshot-calendar-prefix-calendar-header-today"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    Today
  </span>
  <span
    className="snapshot-calendar-prefix-calendar-header-title"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    2018-06-05
  </span>
  <i
    className="icon icon-chevron-right"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  />
</div>
`;
